### PR TITLE
Implemented a way to retrieve consolidated document after update on node

### PR DESCRIPTION
**Pull request by damianr13**
These changes are related to the following FE issue: https://github.com/code4romania/expert-consultation-client/issues/10

### CODE REVIEW

**Comment by damianr13** [comment]
Merged with the last version of develop and implemented tests for the new functionality

File: src/main/java/com/code4ro/legalconsultation/service/impl/DocumentNodeServiceImpl.java [hunk 1]
```diff
@@ -34,6 +34,17 @@ public DocumentNode findById(final UUID id) {
                 .orElseThrow(EntityNotFoundException::new);
     }
 
+    @Transactional(readOnly = true)
+    @Override
+    public DocumentNode findRootNodeForId(UUID id) {
```
Comment (georgebejan) [inline src/main/java/com/code4ro/legalconsultation/service/impl/DocumentNodeServiceImpl.java:39]: We have multiple levels in the node tree. This solution seems to find the root only if the given node id is at the first layer.

Is this intended?

File: src/test/java/com/code4ro/legalconsultation/controller/DocumentNodeControllerTest.java [hunk 2]
```diff
@@ -86,4 +86,9 @@ public void shouldCreateDocumentNodeWithParent() {
 
         assertEquals("parentDocumentNode is different", parentDocumentNode, documentNodeToCreate.getParent());
     }
+
+    @Test
+    public void shouldDeleteDocumentNodeAndReturnRoot() {
```
Comment (georgebejan) [inline src/test/java/com/code4ro/legalconsultation/controller/DocumentNodeControllerTest.java:91]: Please update this test since it does nothing right now

Comment (damianr13) [reply src/test/java/com/code4ro/legalconsultation/controller/DocumentNodeControllerTest.java]: I removed this test method. I started writing it here but realised later that what I wanted actually was an integration test so I moved the logic to `DocumentNodeControllerIntegrationTest` test class

**Pull request merged**